src/Meshes/geometry.ts

```diff
@@ -270,8 +280,9 @@ export class Geometry implements IGetSetVerticesData {
      * @param buffer defines the vertex buffer to use
      * @param totalVertices defines the total number of vertices for position kind (could be null)
      * @param disposeExistingBuffer disposes the existing buffer, if any (default: true)
+     * @param stride defines the stride size to use in the position buffer i.e. the distance between two positions in the positions array. (Default is 3)
      */
-    public setVerticesBuffer(buffer: VertexBuffer, totalVertices: Nullable<number> = null, disposeExistingBuffer = true): void {
+    public setVerticesBuffer(buffer: VertexBuffer, totalVertices: Nullable<number> = null, disposeExistingBuffer = true, stride?: number): void {

```

Comment:
You should not pass the `stride` value to this function: it is a data you can retrieve from `buffer`.`

In `VertexBuffer` you have a `byteStride` property. As it seems you need it as a number of components and not as a number of bytes, you can do `buffer.byteStride / VertexBuffer.GetTypeByteLength(buffer.type)`.